### PR TITLE
scripts: use LIEF for ELF security & symbol checks

### PR DESCRIPTION
**Pull request by fanquake**
This finishes the transition to using LIEF for the ELF symbol and  security checks.

Note that there's currently a work around used for identifying RISCV binaries (just checking the interpreter). I've sent a PR upstream, https://github.com/lief-project/LIEF/pull/562, and we should be able to drop that when using LIEF 0.12.0 and onwards.

### CODE REVIEW

**Review by AHRDcrxvGlEsTBgQwaRHhr254 — APPROVED** [review]

**Comment by practicalswift** [comment]
Concept ACK

**Comment by tryphe** [comment]
Apologies if this information is elsewhere, but could you summarize the advantages, and disadvantages if any, of using LIEF over Wladimir's script?

Does this add a required dependency to the build?

Thank you.

**Comment by fanquake** [comment]
> but could you summarize the advantages,

Bitcoin Core really shouldn't be maintaining it's own ELF parser. This has already proved somewhat problematic, i.e: #21454.

> Does this add a required dependency to the build?

No. It's already needed for release builds, and used for the macOS and Windows binaries.

**Comment by tryphe** [comment]
Concept ACK

**Comment by laanwj** [comment]
Concept ACK.

> Does this add a required dependency to the build?

Not adding a dependency was my reason for writing my own, at the time. But as we already rely on LIEF for windows PE and OSX, using it for ELF too seems fair, it's better than relying on a hodge-podge of text output of various tools, and I'm definitely not planning to write PE and OSX binary parsers.

(I wish I had known of LIEF sooner though, would have saved me some time)

**Comment by fanquake** [comment]
Rebased and un-drafted. Fixed up some outstanding issues in the security checks. This should be ready for review. Now also includes some of the changes from #23148.

**Comment by hebasto** [comment]
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
6c81cb5c88643da842fecd1466ab0112877f91ef5ca558fa11663c546c571a72  guix-build-63b63720bd05/output/aarch64-linux-gnu/SHA256SUMS.part
a1a96d8f4555a87e66b5806f43239f380f49d8107b5479265f7412ff303e5c4b  guix-build-63b63720bd05/output/aarch64-linux-gnu/bitcoin-63b63720bd05-aarch64-linux-gnu-debug.tar.gz
436d1e596fbcb8a00c162a5484db1eed397398bf39af22d065261e53fbdab9a2  guix-build-63b63720bd05/output/aarch64-linux-gnu/bitcoin-63b63720bd05-aarch64-linux-gnu.tar.gz
fd3eda71820a2514007aac4662cb2f162038a8f9d0893c5b8754a2fc54f158c5  guix-build-63b63720bd05/output/arm-linux-gnueabihf/SHA256SUMS.part
61a6401763e0fdda2d046c01e5cc54c7c6e8dc636092c9f7dbfd37413496504e  guix-build-63b63720bd05/output/arm-linux-gnueabihf/bitcoin-63b63720bd05-arm-linux-gnueabihf-debug.tar.gz
18ea82384511a9ea93ee53e520132ddb1d9cac174d22928babef269db7447723  guix-build-63b63720bd05/output/arm-linux-gnueabihf/bitcoin-63b63720bd05-arm-linux-gnueabihf.tar.gz
a34148a28bdbd05c5002fc319f6d57b2e854229ea6077d94178ccbdbad3eb538  guix-build-63b63720bd05/output/dist-archive/bitcoin-63b63720bd05.tar.gz
d0d3177e2fde017ce5286e0310ac75fbbf5a06f65aba05e89f2994eeae64c515  guix-build-63b63720bd05/output/powerpc64-linux-gnu/SHA256SUMS.part
fddd42264d5cf31920185328718a5310a9efec6dd6b8b127ad10098d41dae819  guix-build-63b63720bd05/output/powerpc64-linux-gnu/bitcoin-63b63720bd05-powerpc64-linux-gnu-debug.tar.gz
bc6d969c77e4b5bd8d117081a3a45752441a415fcaf2c6fcf50930794258e6a7  guix-build-63b63720bd05/output/powerpc64-linux-gnu/bitcoin-63b63720bd05-powerpc64-linux-gnu.tar.gz
3e49b0974709c6bc1df2c2e0887534cc82940dad6f37621f19f9712d64b8eb6f  guix-build-63b63720bd05/output/powerpc64le-linux-gnu/SHA256SUMS.part
9e2eb5582074f079712f72fae6b7e65449372a1dc5bb6431ca5394b2d72d0001  guix-build-63b63720bd05/output/powerpc64le-linux-gnu/bitcoin-63b63720bd05-powerpc64le-linux-gnu-debug.tar.gz
1280a0a97653f02da39f91073cd7e65ac402da60546a20e5dd0c0cd8bf27b76c  guix-build-63b63720bd05/output/powerpc64le-linux-gnu/bitcoin-63b63720bd05-powerpc64le-linux-gnu.tar.gz
a8fe290b29e5eca0fcdc4988d730c66e5a02df7fa0a74a9e2a38c9624ed4ad6f  guix-build-63b63720bd05/output/riscv64-linux-gnu/SHA256SUMS.part
b04449be302f6fb5010ad7a221be6c5980a162fa0db5b208d18028a5002acf7f  guix-build-63b63720bd05/output/riscv64-linux-gnu/bitcoin-63b63720bd05-riscv64-linux-gnu-debug.tar.gz
427912609cc80024a093c41c283948905b15afc6da3279a56f414646bd61d805  guix-build-63b63720bd05/output/riscv64-linux-gnu/bitcoin-63b63720bd05-riscv64-linux-gnu.tar.gz
1157360f2c145449e0bc2796dd92d57e0cc7e6735553786cf9ce8285f8b333f1  guix-build-63b63720bd05/output/x86_64-apple-darwin19/SHA256SUMS.part
55032d0ad298e276d61fbe6976dfd4173f32597fce5c8099b1da359f101cc5fe  guix-build-63b63720bd05/output/x86_64-apple-darwin19/bitcoin-63b63720bd05-osx-unsigned.dmg
af4fd797f626008464dba1a03ff70c5be24a163bd2685870242308d71113832c  guix-build-63b63720bd05/output/x86_64-apple-darwin19/bitcoin-63b63720bd05-osx-unsigned.tar.gz
b2a4f43d1da25caf047c05663611bd2521faf68a43fac4970480a00f3c52d648  guix-build-63b63720bd05/output/x86_64-apple-darwin19/bitcoin-63b63720bd05-osx64.tar.gz
7f696f6750b376f832a70912b270425c9fb54ce339d3f5afb808ff6e2a3ee224  guix-build-63b63720bd05/output/x86_64-linux-gnu/SHA256SUMS.part
52776b3b2b8d3945c6cc16bcbc1574ba60adeea141f8a76614b6c39dde1d5614  guix-build-63b63720bd05/output/x86_64-linux-gnu/bitcoin-63b63720bd05-x86_64-linux-gnu-debug.tar.gz
9df46df15f1fd2c1a619c23d550434a37ac7fd2c845c831957d7cc1d8b90b072  guix-build-63b63720bd05/output/x86_64-linux-gnu/bitcoin-63b63720bd05-x86_64-linux-gnu.tar.gz
f3f297b6e17c75e9b0bcfb0100fb678166617f13b29f53de521b2f12c8fcd3f8  guix-build-63b63720bd05/output/x86_64-w64-mingw32/SHA256SUMS.part
ef0bffde3a0de16b40812f65ecf2494d4ae555c56aa613c1bc66cbb983471a1f  guix-build-63b63720bd05/output/x86_64-w64-mingw32/bitcoin-63b63720bd05-win-unsigned.tar.gz
58c930bc308c87b01049704b723520fddba5273e3dc2b1b4b8cee3e70c803ea6  guix-build-63b63720bd05/output/x86_64-w64-mingw32/bitcoin-63b63720bd05-win64-debug.zip
c6eb7174b9dfa516089a29d0780b2d649c4b8d4bf3203b7f05e6bb720b9ceb38  guix-build-63b63720bd05/output/x86_64-w64-mingw32/bitcoin-63b63720bd05-win64-setup-unsigned.exe
1b0f8a91a86bf16396dbf8afd014e9ac4b62cea5fa0024680ed5f2d1ea5bd5b5  guix-build-63b63720bd05/output/x86_64-w64-mingw32/bitcoin-63b63720bd05-win64.zip
```

**Comment by dongcarl** [comment]
Code Review ACK ce69e18947dd917d54eb036e732ac7f7bceede76

**Comment by laanwj** [comment]
Code review ACK ce69e18947dd917d54eb036e732ac7f7bceede76
